Objects/codeobject.c

```diff
@@ -166,6 +166,14 @@ PyCode_NewWithPosOnlyArgs(int argcount, int posonlyargcount, int kwonlyargcount,
         return NULL;
     }
 
+    /* Make sure that code is indexable with an int, this is
+       a long running assumption in ceval.c and many parts of
+       the interpreter. */
+    if (PyBytes_GET_SIZE(code) > INT_MAX) {
+        PyErr_SetString(PyExc_ValueError, "code: co_code is too large");
```

Comment:
Hum, when a limit is caused by the implementation rather than the language specification, usually we use OverflowError rather than ValueError:

```suggestion
        PyErr_SetString(PyExc_OverflowError, "code longer than INT_MAX bytes");
```

Reply:
Aah good catch, thanks!